Library/Formula/telegram.rb

```diff
@@ -0,0 +1,40 @@
+class Telegram < Formula
+  homepage "https://github.com/vysheng/tg"
+  url "https://github.com/vysheng/tg/archive/1.0.5.1.tar.gz"
+  sha1 "72f6a9f89ee9db90b573bad25e567a2015115848"
+
+  depends_on "openssl"
+  depends_on "libevent"
+  depends_on "libconfig"
+  depends_on "readline"
+  depends_on "lua"
+
+  patch :DATA
+
+  def install
+    ENV.append "CFLAGS", "-I#{HOMEBREW_PREFIX}/include -I#{Formula["readline"].prefix}/include"
+    ENV.append "LDFLAGS", "-L#{HOMEBREW_PREFIX}/lib -L#{Formula["readline"].prefix}/lib"
```

Comment:
Unfortunately yes ... you will get the following results if not explicitly modifying CFLAGS and LDFLAGS :-(

```
checking for gcc... clang
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables...
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether clang accepts -g... yes
checking for clang option to accept ISO C89... none needed
checking for sqrt in -lm... yes
checking for library containing clock_gettime... no
checking for library containing backtrace... none required
checking for event_base_new in -levent... yes
checking how to run the C preprocessor... clang -E
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking event2/event.h usability... yes
checking event2/event.h presence... yes
checking for event2/event.h... yes
checking for pkg-config... no
checking for openssl/ssl.h in /usr/local/ssl... no
checking for openssl/ssl.h in /usr/lib/ssl... no
checking for openssl/ssl.h in /usr/ssl... no
checking for openssl/ssl.h in /usr/pkg... no
checking for openssl/ssl.h in /usr/local... no
checking for openssl/ssl.h in /usr... yes
checking whether compiling and linking against OpenSSL works... yes
checking if zlib is wanted... yes
checking for inflateEnd in -lz... yes
checking zlib.h usability... yes
checking zlib.h presence... yes
checking for zlib.h... yes
checking for rl_save_prompt in -lreadline... no
configure: error: no libreadline found
```